api/http.go

```diff
@@ -6,11 +6,14 @@ import (
 	"io/ioutil"
 	"math"
 	"net/http"
+	"sync"
 	"time"
 
 	"github.com/PuerkitoBio/rehttp"
 )
 
+var logMu sync.Mutex
```

Comment:
This could be a field on the `LoggingTransport` struct.

Reply:
good catch!